tests/test_database.py

```diff
@@ -53,3 +55,20 @@ async def test_aio_close_idempotent(db):
 
     await db.aio_close()
     assert db.is_connected is False
+
+
+@pytest.mark.parametrize('db_name', PG_DBS + MYSQL_DBS)
+async def test_deferred_init(db_name):
+    init_params = DB_DEFAULTS[db_name]
+    database_host = init_params.pop('database')
+    init_params['database'] = None
+
+    database: AioDatabase = DB_CLASSES[db_name](**init_params)
+
+    with pytest.raises(Exception, match='Cannot create connection before db inited'):
+        await database.aio_execute_sql(sql='SELECT 1;')
+
+    init_params['database'] = database_host
+    database.init(**init_params)
+
+    await database.aio_connect()
```

Comment:
Let's change it to something like database.aio_execute_sql(sql='SELECT 1;') since aio_connect rarely used